sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/kinesis/enhancedfanout/EFOShardSubscribersPool.java

```diff
@@ -387,18 +410,15 @@ public EventRecords(String streamName, String shardId, SubscribeToShardEvent eve
 
     @Override
     protected Iterator<KinesisClientRecord> delegate() {
-      if (event.hasRecords() && !event.records().isEmpty()) {
```

Comment:
The logical order of if clauses was reversed here, not a big problem but this should reflect better that delegate is lazily initialized.